### PR TITLE
[Bug fix] Export all branches for discrete control torch

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

When exporting the Torch Policy, we will now export all branches and not only the first one.
This should fix the inference regression on FoodCollector

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by chriselion** [comment]
Should we apply this on the release branch? It won't require rerunning the yamato step, so the impact is minimal.

**Comment by vincentpierre** [comment]
> Should we apply this on the release branch? It won't require rerunning the yamato step, so the impact is minimal.

Sure, I can make another PR or we can cherry pick after merge. What do you say, how do you want to do this?

**Comment by chriselion** [comment]
"cherry pick after merge" sounds good. I don't feel qualified to review the actual change though.

File: ml-agents/mlagents/trainers/torch/networks.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def __init__(
         self.is_continuous_int = torch.nn.Parameter(
             torch.Tensor([int(act_type == ActionType.CONTINUOUS)])
         )
-        self.act_size_vector = torch.nn.Parameter(torch.Tensor(act_size))
+        self.act_size_vector = torch.nn.Parameter(torch.Tensor([sum(act_size)]))
```
Comment (dongruoping) [inline ml-agents/mlagents/trainers/torch/networks.py:270]:
```suggestion
        self.act_size_vector = torch.nn.Parameter(torch.Tensor([sum(act_size)]), requires_grad=False)
```

**Review by dongruoping — APPROVED** [review]
looks good to me

File: ml-agents/mlagents/trainers/torch/networks.py [hunk 2]
```diff
@@ -329,12 +329,12 @@ def forward(
         Note: This forward() method is required for exporting to ONNX. Don't modify the inputs and outputs.
         """
         dists, _ = self.get_dists(vec_inputs, vis_inputs, masks, memories, 1)
-        action_list = self.sample_action(dists)
-        sampled_actions = torch.stack(action_list, dim=-1)
         if self.act_type == ActionType.CONTINUOUS:
+            action_list = self.sample_action(dists)
+            sampled_actions = torch.stack(action_list, dim=-1)
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/torch/networks.py:334]: why not just `action_out = torch.stack(action_list, dim=-1)`

File: ml-agents/mlagents/trainers/torch/networks.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/torch/networks.py:270]: should this be a `buffer`?

**Review by ervteng — APPROVED** [review]
🐛 🔧

File: ml-agents/mlagents/trainers/torch/networks.py (hunk 1 above)
Comment (dongruoping) [reply ml-agents/mlagents/trainers/torch/networks.py:270]: I remember I left it as parameter due to some errors I saw during export using buffer. Now sure if it's still the case.

File: ml-agents/mlagents/trainers/torch/networks.py [hunk 3]
```diff
@@ -329,12 +329,12 @@ def forward(
...
             action_out = sampled_actions
         else:
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/torch/networks.py:336]:
```suggestion
            action_out = torch.stack(action_list, dim=-1)
        else:
```
